x/genutil/types/genesis_state.go

```diff
@@ -0,0 +1,32 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+
+	tmos "github.com/tendermint/tendermint/libs/os"
+)
+
+// GenesisStateFromGenFile creates the core parameters for genesis initialization
+// for the application.
+//
+// NOTE: The pubkey input is this machines pubkey.
+func GenesisStateFromGenFile(genFile string) (genesisState map[string]json.RawMessage, genDoc map[string]interface{}, err error) {
```

Comment:
if you do concrete type assertions without checking `ok` on the genesisState, can we have a concrete type instead?